helm-chart/renku-notebooks/values.yaml

```diff
@@ -268,3 +268,7 @@ nodeSelector: {}
 tolerations: []
 
 affinity: {}
+
+# Restricts egress from the jupyterlab user pods only to DNS (internal and external), external IPs
+# and the jupyterhub service pods, setting this to false results in fully unrestricted egress.
+restrictUserPodEgress: false
```

Comment:
Network policies might be very specific to the service in question - e.g. we have a network policy for redis, I wouldn't want the same switch controlling that policy and this one. So I think the way it is now in this PR is fine, it's explicitly about the user jupyter server pods.

Reply:
>e.g. we have a network policy for redis, I wouldn't want the same switch controlling that policy and this one

Yes, the redis network policies are problematic and should - at least for the time being - be always on, no matter what. 

In general I would just like to not add too many configuration options to the charts and instead refer to post-rendering options for infrequent use cases. But since this network policy is indeed quite special (and different from the ingress network policies that we are hopefully going to add for our other components in the future) I'm fine with leaving this one separate. But then I would prefer to set the default to `true`.

Reply:
I think we should follow the rest of the k8s/helm convention and default to open rather than closed. Production values should not be the default imho.

Reply:
I wouldn't consider this particular network policy a production setting. For me it's a specific security setting that should be on by default, just like not running user pods as root.
I know that most helm charts turn network policies off by default, but those are often network policies that restrict all communication that is not explicitly allowed.